storm-core/src/jvm/org/apache/storm/metrics2/StormMetricRegistry.java

```diff
@@ -0,0 +1,135 @@
+        return new DisruptorMetrics(
+                StormMetricRegistry.gauge(0L, name + "-capacity", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-population", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-write-position", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-read-position", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0.0, name + "-arrival-rate", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0.0, name + "-sojourn-time-ms", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0L, name + "-overflow", topologyId, componentId, port),
+                StormMetricRegistry.gauge(0.0F, name + "-percent-full", topologyId, componentId, port)
+        );
+    }
+
+    public static Meter meter(String name, WorkerTopologyContext context, String componentId){
+        String metricName = metricName(name, context.getStormId(), componentId, context.getThisWorkerPort());
+        return REGISTRY.meter(metricName);
+    }
+
+    public static void start(Map<String, Object> stormConfig, DaemonType type){
+        String localHost = (String)stormConfig.get(Config.STORM_LOCAL_HOSTNAME);
+        if(localHost != null){
+            hostName = localHost;
+        } else {
+            try {
+                hostName = InetAddress.getLocalHost().getCanonicalHostName();
+            } catch (UnknownHostException e) {
+                 LOG.warn("Unable to determine hostname while starting the metrics system. Hostname ill be reported" +
+                         " as 'localhost'.");
+            }
+        }
+
+        LOG.info("Starting metrics reporters...");
+        List<Map<String, Object>> reporterList = (List<Map<String, Object>>)stormConfig.get(Config.STORM_METRICS_REPORTERS);
+        if(reporterList != null && reporterList.size() > 0) {
+            for (Map<String, Object> reporterConfig : reporterList) {
+                // only start those requested
+                List<String> daemons = (List<String>) reporterConfig.get("daemons");
+                for (String daemon : daemons) {
+                    if (DaemonType.valueOf(daemon.toUpperCase()) == type) {
+                        startReporter(stormConfig, reporterConfig);
+                    }
+                }
+            }
+        }
+    }
+
+    public static MetricRegistry registry(){
+        return REGISTRY;
+    }
+
+    private static void startReporter(Map<String, Object> stormConfig, Map<String, Object> reporterConfig){
+        String clazz = (String)reporterConfig.get("class");
+        StormReporter reporter = null;
+        LOG.info("Attempting to instantiate reporter class: {}", clazz);
+        try{
+            reporter = (StormReporter)Metrics2Utils.instantiate(clazz);
```

Comment:
Unfortunately we can't. `Utils.newInstance()` will throw a `RuntimeException` if the class fails to load. Here we just want to catch any exceptions, log a warning, and skip that reporter, rather than crash.

Reply:
I'm not sure I understand why we don't want to crash in that case? Won't that only happen due to misconfiguration?

Reply:
If we don't want to crash for a mis-configuration we should at least include a comment and explain why we are not crashing.

Reply:
My reasoning was that a metrics reporter misconfiguration isn't critical -- Storm will continue its core function (running topologies). Mainly a judgement call.

I'll change it to crash instead.